Utilities/File.cpp

```diff
@@ -1550,12 +1550,20 @@ void fs::remove_all(const std::string& path, bool remove_root)
 
 		if (entry.is_directory == false)
 		{
-			remove_file(path + '/' + entry.name);
+			if (!remove_file(path + '/' + entry.name))
+			{
+				return;
+			}
 		}
 
 		if (entry.is_directory == true)
 		{
 			remove_all(path + '/' + entry.name);
+
+			if (g_tls_error != fs::error::ok)
```

Comment:
It won't really work since g_tls_error is not reset on success.

Reply:
That also means making remove_all return initial success status (bool) is needed for proper error handling.